components/aws/sagemaker/common/_utils.py

```diff
@@ -113,17 +122,49 @@ def print_logs_for_job(cw_client, log_grp, job_name):
         logging.error(e)
 
 
-def get_sagemaker_client(region, endpoint_url=None):
+def get_boto3_session(region, role_arn=None):
+    """Creates a boto3 session, optionally assuming a role"""
+
+    # By default return a basic session
+    if role_arn is None:
+        return Session()
+
+    def refresh():
+        client = boto3.client("sts", region_name=region,
+                                endpoint_url='https://sts.' + region + '.amazonaws.com')
+        response = client.assume_role(RoleArn=role_arn, RoleSessionName=ASSUME_SESSION_NAME_FORMAT).get("Credentials")
+        return dict(
+            access_key=response.get("AccessKeyId"),
+            secret_key=response.get("SecretAccessKey"),
+            token=response.get("SessionToken"),
+            expiry_time=response.get("Expiration").isoformat()
+        )
+
+    session_credentials = RefreshableCredentials.create_from_metadata(
+        metadata=refresh(),
+        refresh_using=refresh,
+        method='sts-assume-role'
+    )
+
+    current_session = get_session()
+    # Must modify this internally, no clean way to set this otherwise
+    current_session._credentials = session_credentials
```

Comment:
whats going on here, can you point me to some documentation to refer to

Reply:
Taken from the example documents listed above. I'm not entirely sure why this is necessary.